Koha/Plugin/Fi/KohaSuomi/DI/PatronController.pm

```diff
@@ -553,26 +554,46 @@ sub validate_credentials {
     }
 
     my $dbh = C4::Context->dbh;
-    unless (C4::Auth::checkpw_internal($dbh, $userid, $password)) {
+    
+    my $patron = Koha::Patrons->find({ userid => $userid });
+    $patron = Koha::Patrons->find({ cardnumber => $userid }) unless $patron;
+    
+    if (!$patron) {
         return $c->render(
             status => 401, 
             openapi => { error => "Login failed." }
         );
     }
-
-    my $patron = Koha::Patrons->find({ userid => $userid });
-    $patron = Koha::Patrons->find({ cardnumber => $userid }) unless $patron;
-
-    if ($patron && $patron->lost) {
-        return $c->render( 
-            status => 403, 
-            openapi => { 
-                error => "Patron's card has been marked as 'lost'. Access forbidden." 
-            }
+    
+    if ($patron->account_locked || !C4::Auth::checkpw_internal($dbh, $userid, $password)) {
```

Comment:
this should be:
```Perl
    if ($patron->account_locked || !(
            C4::Context->preference('Version') ge '22.110000'
                ? C4::Auth::checkpw_internal($userid, $password)
                : C4::Auth::checkpw_internal($dbh, $userid, $password)
        )) {
```
because in 22.11: `Bug 27342: Remove dbh from C4::Auth`
also refer here: https://github.com/NatLibFi/koha-plugin-rest-di/pull/20

Reply:
This is now committed, so rebase needed.

Reply:
yes, and I posted changes "here and there" to get attention from @lmstrand to not to lose these important changes/compatibility. Thanks!